custom_components/elasticsearch/config_flow.py

```diff
@@ -306,9 +319,12 @@ async def async_step_import(self, import_config):
             return self.async_abort(reason="single_instance_allowed")
 
         self.config = build_full_config(import_config)
-        (success, errors) = await self._async_elasticsearch_login()
+        # Configure legacy yml to use legacy index mode
+        self.config[CONF_INDEX_MODE] = INDEX_MODE_LEGACY
```

Comment:
The call to `build_full_config` was defaulting the legacy yml import to use datastreams. I'm adjusting this flow to ensure that legacy configs continue to use the legacy index mode.